runtime-sdk/src/error.rs

```diff
@@ -29,26 +29,171 @@ use crate::types::transaction::CallResult;
 /// ```
 pub trait Error: std::error::Error {
     /// Name of the module that emitted the error.
-    fn module_name() -> &'static str;
+    fn module_name(&self) -> &'static str;
```

Comment:
Yeah, I agree that the other syntax is a bit awkward. It's probably safe to assume that nobody will ever work with an error independently of an instance (c.f. marker types).